Exposed relocation reflection issue

### Expected and Results

When relocating libraries, Exposed having an issue with it's reflection it seems and can't find kotlin.Any

### Related environment and versions

Gradle 8.7

ShadowJar 9.0.0-beta12

Exposed 0.50.1

Kotlin 2.2.0

Java 21

OS: Linux(Debian/Arch)/MacOs

### Reproduction steps

I have a whole project which reproduces this issue https://github.com/makeevrserg/ExposedGradleFail

### Anything else?

_No response_

Seems due to #843.

We can't apply the workaround like:

```kt
  relocate("", requireProjectInfo.group) {
    exclude("META-INF/**")
    exclude("kotlin/**")
    exclude("kotlinx/**")
  }
```

as there will be a new issue like:

```
  at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.visitFile(ShadowCopyAction.kt:167)
  at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.processFile(ShadowCopyAction.kt:152)
  ... 159 more
Caused by: java.lang.IllegalArgumentException: UTF8 string too large
  at org.objectweb.asm.ByteVector.encodeUtf8(ByteVector.java:307)
  at org.objectweb.asm.ByteVector.putUTF8(ByteVector.java:274)
  at org.objectweb.asm.SymbolTable.addConstantUtf8(SymbolTable.java:800)
  at org.objectweb.asm.AnnotationWriter.visit(AnnotationWriter.java:198)
  at org.objectweb.asm.AnnotationVisitor.visit(AnnotationVisitor.java:111)
  at org.objectweb.asm.commons.AnnotationRemapper.visit(AnnotationRemapper.java:113)
  at org.objectweb.asm.ClassReader.readElementValue(ClassReader.java:3083)
  at org.objectweb.asm.ClassReader.readElementValues(ClassReader.java:3012)
  at org.objectweb.asm.ClassReader.readElementValue(ClassReader.java:3186)
  at org.objectweb.asm.ClassReader.readElementValues(ClassReader.java:3006)
  at org.objectweb.asm.ClassReader.accept(ClassReader.java:609)
  at org.objectweb.asm.ClassReader.accept(ClassReader.java:425)
  at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction$StreamAction.remapClass(ShadowCopyAction.kt:199)
  ... 161 more
```

I refactored the logic and tested your reproducer

<img width="800" height="363" alt="Image" src="https://github.com/user-attachments/assets/c7a8a0a0-1b93-40fd-a868-50291ae0cd95" />

The key point is 

```diff
shadowJar.configure {
    // ...

--    relocationPrefix = requireProjectInfo.group
--    enableAutoRelocation = true

++    relocate("kotlin/", "example/kotlin/") {
++        exclude("kotlin/kotlin.kotlin_builtins")
++    }
}
```

We have to exclude relocating `kotlin/kotlin.kotlin_builtins`, this is why 

```
Exception in thread "main" java.lang.AssertionError: Built-in class com.makeevrserg.exposedgradlefail.kotlin.Any is not found
  at com.makeevrserg.exposedgradlefail.kotlin.reflect.jvm.internal.impl.builtins.KotlinBuiltIns$3.invoke(KotlinBuiltIns.java:93)
  at com.makeevrserg.exposedgradlefail.kotlin.reflect.jvm.internal.impl.builtins.KotlinBuiltIns$3.invoke(KotlinBuiltIns.java:88)
  at com.makeevrserg.exposedgradlefail.kotlin.reflect.jvm.internal.impl.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
  at com.makeevrserg.exposedgradlefail.kotlin.reflect.jvm.internal.impl.storage.LockBasedStorageManager$MapBasedMemoizedFunctionToNotNull.invoke(LockBasedStorageManager.java:681)
  at com.makeevrserg.exposedgradlefail.kotlin.reflect.jvm.internal.impl.builtins.KotlinBuiltIns.getBuiltInClassByName(KotlinBuiltIns.java:223)
  at com.makeevrserg.exposedgradlefail.kotlin.reflect.jvm.internal.impl.builtins.KotlinBuiltIns.getAny(KotlinBuiltIns.java:228)
```

--- a/docs/configuration/relocation/README.md
+++ b/docs/configuration/relocation/README.md
@@ -210,6 +210,13 @@ To configure automatic dependency relocation, set `enableAutoRelocation = true` 
 > Be mindful that some Gradle plugins will automatically add dependencies to your class path. You may need to remove these
 > dependencies if you do not intend to shadow them into your library.
 
+
+## Relocating Kotlin Standard Library
+
+It is not recommended to relocate Kotlin Standard Library if you are using [Kotlin Metadata][kotlin-metadata] or
+[Kotlin Reflection][kotlin-reflection] in the project, because they are tightly coupled with Kotlin compiler and
+runtime. See more details and discussion in [#1622][#1622].
+
 ## Relocating Project Resources Only
 
 If you want to relocate the resources of the project only and exclude all dependencies (related to a normal JAR but with
@@ -237,3 +244,9 @@ relocating), you can try out the trick like:
 
 This is useful in some cases like [#759](https://github.com/GradleUp/shadow/issues/759) mentioned. See
 [Configuring Shadowed Dependencies](../dependencies/README.md) for more information about `configurations`.
+
+
+
+[#1622]: https://github.com/GradleUp/shadow/issues/1622
+[kotlin-metadata]: https://kotlinlang.org/docs/metadata-jvm.html
+[kotlin-reflection]: https://kotlinlang.org/docs/reflection.html

